apps/examples/control_log/ctrlog.c

```diff
@@ -0,0 +1,371 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+#include <tinyara/net/netlog.h>
+#include <tinyara/net/netlog_lwip.h>
+#include "ctrlog_utils.h"
+#include "ctrlog_usage.h"
+
+#define CMD_LEVEL "level="
+#define CMD_TIME "time="
+#define CMD_FUNC "func="
+#define CMD_FILE "file="
+#define CMD_COLOR "color="
+#define CMD_MOD "mod="
+#define CMD_RESET "reset"
+
+static char *g_lwip_mod_table_str[] = {
+		"etharp",
+		"netif",
+		"pbuf",
+		"api_lib",
+		"api_msg",
+		"sockets",
+		"icmp",
+		"igmp",
+		"inet",
+		"ip",
+		"ip_reass",
+		"raw",
+		"mem",
+		"memp",
+		"sys",
+		"timers",
+		"tcp",
+		"tcp_input",
+		"tcp_fr",
+		"tcp_rto",
+		"tcp_cwnd",
+		"tcp_wnd",
+		"tcp_output",
+		"tcp_rst",
+		"tcp_qlen",
+		"udp",
+		"tcpip",
+		"slip",
+		"dhcp",
+		"autoip",
+		"dns",
+		"ip6",
+		"nd6"};
+
+static uint32_t g_lwip_mod_table[] = {
+		ETHARP_DEBUG,
+		NETIF_DEBUG,
+		PBUF_DEBUG,
+		API_LIB_DEBUG,
+		API_MSG_DEBUG,
+		SOCKETS_DEBUG,
+		ICMP_DEBUG,
+		IGMP_DEBUG,
+		INET_DEBUG,
+		IP_DEBUG,
+		IP_REASS_DEBUG,
+		RAW_DEBUG,
+		MEM_DEBUG,
+		MEMP_DEBUG,
+		SYS_DEBUG,
+		TIMERS_DEBUG,
+		TCP_DEBUG,
+		TCP_INPUT_DEBUG,
+		TCP_FR_DEBUG,
+		TCP_RTO_DEBUG,
+		TCP_CWND_DEBUG,
+		TCP_WND_DEBUG,
+		TCP_OUTPUT_DEBUG,
+		TCP_RST_DEBUG,
+		TCP_QLEN_DEBUG,
+		UDP_DEBUG,
+		TCPIP_DEBUG,
+		SLIP_DEBUG,
+		DHCP_DEBUG,
+		AUTOIP_DEBUG,
+		DNS_DEBUG,
+		IP6_DEBUG,
+		ND6_DEBUG,
+};
+
+static char *g_level_table_str[] = {
+		"error",
+		"info",
+		"verb",
+};
+
+static uint32_t g_level_table[] = {
+		NL_LEVEL_ERROR,
+		NL_LEVEL_INFO,
+		NL_LEVEL_VERB,
+};
+
+static char *g_color_table_str[] = {
+		"red", "yellow", "green", "default"};
+
+static nl_color_e g_color_table[] = {
+		NL_COLOR_RED, NL_COLOR_YELLOW, NL_COLOR_GREEN, NL_COLOR_GREEN};
+
+static char *g_mod_table_str[] = {
+		"wifi", "netlib", "lwip", "netmgr"};
+
+static netlog_module_e g_mod_table[] = {
+		NL_MOD_WIFI_MANAGER,
+		NL_MOD_NETLIB,
+		NL_MOD_LWIP,
+		NL_MOD_NET_MANAGER,
+};
+
+static char *g_opt_table_str[] = {
+		"on",
+		"off",
+};
+
+static nl_options g_opt_table[] = {
+		NL_OPT_ENABLE,
+		NL_OPT_DISABLE,
+};
+
+static uint32_t _get_lwip_mod(char *mod_str)
+{
+	int size = sizeof(g_lwip_mod_table_str) / sizeof(char *);
+	for (int i = 0; i < size; i++) {
+		if ((strlen(mod_str) == strlen(g_lwip_mod_table_str[i])) && strncmp(mod_str, g_lwip_mod_table_str[i], strlen(mod_str) + 1) == 0) {
```

Comment:
How about declaring local variable for `strlen(mod_str)`?
Now, we should calculate the length everytime, even for `strncmp`.
If we declare one local variable and use it, we can optimize the code.

Reply:
Thanks for your opinion. I also considered the case you mentioned. But I thought that the function wouldn't be called frequently. Also strlen is called twice. So I thought that it's acceptable.